src/cljs/ataru/virkailija/application/application_search_control.cljs

```diff
@@ -21,35 +21,34 @@
     (str " (" count ")")
     ""))
 
-(defn ssn-search-field []
-  (let [ssn-value (subscribe [:state-query [:application :search-control :ssn :value]])]
+(defn search-term-field []
+  (let [search-term (subscribe [:state-query [:application :search-control :search-term :value]])]
     [:div
-     [:input.application__search-control-ssn-input
+     [:input.application__search-control-search-term-input
       {:type        "text"
        :auto-focus  true
        :id          "ssn-search-field"
-       :class       (when (true? @(subscribe [:state-query [:application :search-control :ssn :show-error]]))
-                      "application__search-control-ssn-input-error animated shake")
-       :placeholder "Etsi henkilötunnuksella"
-       :value       @ssn-value
-       :max-length  "11"
-       :on-change   (fn [evt] (dispatch [:application/ssn-search (-> evt .-target .-value)]))}]
-     (when-not (clojure.string/blank? @ssn-value)
-       [:span.application__search-control-clear-ssn
+       :class       (when (true? @(subscribe [:state-query [:application :search-control :search-term :show-error]]))
+                      "application__search-control-search-term-input-error animated shake")
+       :placeholder "Etsi henkilötunnuksella, syntymäajalla tai sähköpostiosoitteella"
+       :value       @search-term
+       :on-change   (fn [evt] (dispatch [:application/search-by-term (-> evt .-target .-value)]))}]
+     (when-not (clojure.string/blank? @search-term)
+       [:span.application__search-control-clear-search-term
         {:on-click #(dispatch [:application/clear-applications-haku-and-form-selections])}
         [:i.zmdi.zmdi-close]])]))
 
-(defn search-ssn-tab [tab-id selected-tab link-url label-text]
+(defn search-term-tab [tab-id selected-tab link-url label-text]
   (let [tab-selected (when (= tab-id selected-tab) "application__search-control-selected-tab-with-input")]
     [:div.application__search-control-tab-selector-wrapper
      [:a {:href link-url
           :on-click (fn [event]
```

Comment:
This click-handler is called even when the search field is already visible. In other words, doing the following

1. Clicking on the "Etsi henkilötunnuksella..." button
2. Inputting e.g. a birthdate "1.1.2000" to the field which appear in place of the button
3. Clicking on the search results (or in any way blurring the input field)
4. Clicking back on the input field

.. resets the search field and also clears any search results. The reason seems to be that the button click handler (from step 1) is triggered even though the search field is already visible (which triggers the `/applications/search/` front route, resetting the state).